### PR TITLE
[TF][Spec] Conv2d, AtrousConv2d, DepthwiseConv2d, SeparableConv2d, DepthToSpace, SpaceToDepth, ExpandDims

### PR DESCRIPTION
**Pull request by Kristoff-starling**
Add the following operators for TensorFlow:
* Conv2d (switch to functional API)
* AtrousConv2d
* DepthwiseConv2d
* SeparableConv2d
* DepthToSpace
* SpaceToDepth
* ExpandDims

### CODE REVIEW

File: nnsmith/materialize/tensorflow/dialect.py [hunk 1]
```diff
@@ -96,66 +98,75 @@ def requires(self, input_shapes: List[AbsTensor]) -> List[z3.ExprRef]:
...
 
 
-class NHWCConv2d(UnaryOpBase):
-    in_dtypes = [(DType.float16,), (DType.float32,), (DType.float64,)]
+class NHWCConv2dBase(BinaryOpBase):
+    in_dtypes = [
+        (DType.float16, DType.float16),
+        (DType.float32, DType.float32),
+        (DType.float64, DType.float64),
+    ]
```
Comment (ganler) [inline nnsmith/materialize/tensorflow/dialect.py:106]:
```suggestion
    in_dtypes = DTYPE_GEN_FLOATS
```

File: nnsmith/materialize/tensorflow/dialect.py [hunk 2]
```diff
@@ -181,6 +192,19 @@ def requires(self, input_shapes):
         )
         return cons
 
+    def flops(self, input_shapes):
```
Comment (ganler) [inline nnsmith/materialize/tensorflow/dialect.py:195]: `flops` and `n_floats` are no longer needed so we can trim them.

**Review by ganler — CHANGES_REQUESTED** [review]

File: nnsmith/materialize/tensorflow/dialect.py (hunk 1 above)
Comment (Kristoff-starling) [reply nnsmith/materialize/tensorflow/dialect.py]: `BinaryOpBase` has defined the `in_dtypes` and `out_dtypes` so we needn't redefine them here. Will remove them in the next commit.

Comment (ganler) [reply nnsmith/materialize/tensorflow/dialect.py]: true. also i just noticed my prior suggestion is kinda wrong...

**Review by ganler — APPROVED** [review]
LGTM

**Pull request merged**